From 849fbc549d4cca576d659d7df139c5f04104cb48 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Tue, 29 Jun 2021 09:24:31 +0900
Subject: [PATCH] ksmbd: opencode to remove ATTR_FP macro
Git-commit: 849fbc549d4cca576d659d7df139c5f04104cb48
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Opencode to remove ATTR_FP macro.

Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ksmbd/oplock.c    | 4 +++-
 fs/ksmbd/vfs_cache.h | 5 -----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/fs/ksmbd/oplock.c b/fs/ksmbd/oplock.c
index 3f0dd9b35c78..43c8b7ce6095 100644
--- a/fs/ksmbd/oplock.c
+++ b/fs/ksmbd/oplock.c
@@ -1111,7 +1111,9 @@ int smb_grant_oplock(struct ksmbd_work *work, int req_op_level, u64 pid,
 		goto set_lev;
 
 	/* grant none-oplock if second open is trunc */
-	if (ATTR_FP(fp)) {
+	if (fp->attrib_only && fp->cdoption != FILE_OVERWRITE_IF_LE &&
+	    fp->cdoption != FILE_OVERWRITE_LE &&
+	    fp->cdoption != FILE_SUPERSEDE_LE) {
 		req_op_level = SMB2_OPLOCK_LEVEL_NONE;
 		goto set_lev;
 	}
diff --git a/fs/ksmbd/vfs_cache.h b/fs/ksmbd/vfs_cache.h
index 752cbdab3522..543494f664cb 100644
--- a/fs/ksmbd/vfs_cache.h
+++ b/fs/ksmbd/vfs_cache.h
@@ -25,11 +25,6 @@
 #define KSMBD_NO_FID		(UINT_MAX)
 #define SMB2_NO_FID		(0xFFFFFFFFFFFFFFFFULL)
 
-#define ATTR_FP(fp) ((fp)->attrib_only && \
-		     ((fp)->cdoption != FILE_OVERWRITE_IF_LE && \
-		      (fp)->cdoption != FILE_OVERWRITE_LE && \
-		      (fp)->cdoption != FILE_SUPERSEDE_LE))
-
 struct ksmbd_conn;
 struct ksmbd_session;
 
-- 
2.33.0

